src/bin/ion/commands/generate/generator.rs

```diff
@@ -656,6 +668,122 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
 
         Ok(AbstractDataType::Scalar(scalar_builder.build()?))
     }
+
+    /// Builds `AbstractDataType::WrappedSequence` from the given constraints.
+    /// ```
+    /// type::{
+    ///   name: foo,
+    ///   type: list,
+    ///   element: string,
+    /// }
+    /// ```
+    /// This method builds `AbstractDataType`as following:
+    /// ```
+    /// AbstractDataType::WrappedSequence(
+    ///  WrappedSequence {
+    ///     name: vec!["org", "example", "Foo"] // assuming the namespace here is `org.example`
+    ///     element_type: FullyQualifiedTypeReference { type_name: vec!["String"], parameters: vec![] } // Represents the element type for the list
+    ///     sequence_type: SequenceType::List, // Represents list type for the given sequence
+    ///     doc_comment: None // There is no doc comment defined in above ISL type def
+    ///     source: IslType { .. } // Represents the `IslType` that is getting converted to `AbstractDataType`
+    ///  }
+    /// )
+    /// ```
+    fn build_wrapped_sequence_from_constraints(
+        &mut self,
+        constraints: &[IslConstraint],
+        code_gen_context: &mut CodeGenContext,
+        parent_isl_type: &IslType,
+    ) -> CodeGenResult<AbstractDataType> {
+        let mut wrapped_sequence_builder = WrappedSequenceBuilder::default();
+        wrapped_sequence_builder
+            .name(self.current_type_fully_qualified_name.to_owned())
+            .source(parent_isl_type.to_owned());
+        for constraint in constraints {
+            match constraint.constraint() {
+                IslConstraintValue::Element(isl_type_ref, _) => {
```

Comment:
Please make sure to return an error if there are two of the `element` constraint (so that we don't have to deal with potential contradictions).